.github/workflows/tests.yml

```diff
@@ -119,3 +119,22 @@ jobs:
         with:
           command: clippy
           args: -- -D warnings
+
+  publish:
+    name: Publish Package
+    needs: [check, lints, test]
+    if: startsWith(github.ref, 'refs/tags/v')
```

Comment:
The tags we use for now don't start with `v`, so either
- we start using `v{MAJOR}.{MINOR}.{PATCH}` now, or
- we match against `refs/tags/` instead

I don't have any strong opinion on the matter.